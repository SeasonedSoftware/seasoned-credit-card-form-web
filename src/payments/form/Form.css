.distance-fields {
  max-width: 400px;
  margin: 0 auto;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
}

.distance-fields fieldset {
  vertical-align: top;
}

.distance-fields button {
  margin-top: 20px;
}

.credit-card {
  margin: 20px auto;
}

#value-fieldset input {
  padding-left: 25px;
}

#value-fieldset::before {
  content: 'R$';
  left: 0;
  position: absolute;
  top: 20px;
  font-family: 'Open Sans';
  color: #888;
}

@media (max-width: 425px) {
  .distance-fields {
    display: flex;
    flex-direction: column;
  }

  .distance-fields button {
    position: fixed;
    bottom: 0;
    display: block;
    margin: 0;
    box-sizing: border-box;
    left: 0;
    z-index: 2;
  }

  .distance-fields fieldset {
    flex: 1;
  }

  fieldset {
    order: 0;
  }

  .credit-card {
    order: 1;
  }

  .dialog {
    padding: 10px;
  }
}
