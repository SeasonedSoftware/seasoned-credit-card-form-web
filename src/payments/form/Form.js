import React, { Component } from 'react'
import filter from 'lodash/filter'
import { Form } from 'formik'
import Grid from '@material-ui/core/Grid'
import Cards from 'react-credit-cards'

import Error from '../../Error'
import Loading from '../../Loading'
import Button from './Button'
import Field from './Field'
import fields from './fields'
import onFocus from './onFocus'

export default (class extends Component {
  constructor(props) {
    super(props)

    this.state = {
      focused: '',
    }

    this.onFocus = onFocus.bind(this)
  }

  commonProperties() {
    return {
      touched: this.props.touched,
      errors: this.props.errors,
      handleChange: this.props.handleChange,
      handleBlur: this.props.handleBlur,
      setFieldValue: this.props.setFieldValue,
      onFocus: target => this.onFocus(target),
    }
  }

  renderFields() {
    return (
      <Grid container justify={'space-between'} style={{ margin: '30px 0' }}>
        {filter(fields(this.props)).map(element => {
          return (
            <Field
              key={element.name}
              type={element.type || 'text'}
              value={this.props.values[element.name]}
              {...element}
              {...this.commonProperties()}
            />
          )
        })}
      </Grid>
    )
  }

  render() {
    const { error, submitting } = this.props

    return (
      <Form>
        <div>
          <Cards
            number={this.props.values.number}
            name={this.props.values.name}
            expiry={this.props.values.expiry}
            cvc={this.props.values.cvc}
            focused={this.state.focused}
            placeholders={{ name: 'Seu nome' }}
          />
        </div>
        {this.renderFields()}
        {error && <Error>{error}</Error>}
        {submitting && <Loading style={{ alignSelf: 'center' }} />}
        <Button disabled={submitting} />
      </Form>
    )
  }
})
