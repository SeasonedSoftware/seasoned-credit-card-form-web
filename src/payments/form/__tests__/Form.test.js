import React from 'react'
import renderer from 'react-test-renderer'

import Form from '../Form'

jest.mock('../Field', () => props => <div {...props}>Field</div>)

jest.mock('formik', () => ({
  Form: props => <div {...props}>{props.children}</div>,
  Field: props => <div {...props}>{props.children}</div>,
}))

const values = {
  name: '',
  number: '',
  expiry: '',
  cvc: '',
}

const props = {
  handleChange: jest.fn(),
  errors: {},
  touched: {},
}

const tree = renderer.create(<Form {...props} values={values} />)
it('renders correctly with one card', () => {
  expect(tree.toJSON()).toMatchSnapshot()
})

it('constructor puts focused as a state property', () => {
  expect(tree.getInstance().state).toHaveProperty('focused')
})

it('binded onFocus change his state', () => {
  const target = { id: 'powpow' }
  let instance = tree.getInstance()
  expect(instance.state.focused).toEqual('')
  instance.onFocus({ target })
  expect(instance.state.focused).toEqual(target.id)
})
