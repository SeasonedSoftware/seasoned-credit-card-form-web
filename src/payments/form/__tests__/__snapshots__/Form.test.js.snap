// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with one card 1`] = `
<div>
  <div>
    <div
      className="rccs"
    >
      <div
        className="rccs__card rccs__card--unknown"
      >
        <div
          className="rccs__card--front"
        >
          <div
            className="rccs__card__background"
          />
          <div
            className="rccs__issuer"
          />
          <div
            className="rccs__cvc__front"
          >
            
          </div>
          <div
            className="rccs__number"
          >
            •••• •••• •••• ••••
          </div>
          <div
            className="rccs__name"
          >
            Seu nome
          </div>
          <div
            className="rccs__expiry"
          >
            <div
              className="rccs__expiry__valid"
            >
              valid thru
            </div>
            <div
              className="rccs__expiry__value"
            >
              ••/••
            </div>
          </div>
          <div
            className="rccs__chip"
          />
        </div>
        <div
          className="rccs__card--back"
        >
          <div
            className="rccs__card__background"
          />
          <div
            className="rccs__stripe"
          />
          <div
            className="rccs__signature"
          />
          <div
            className="rccs__cvc"
          >
            
          </div>
          <div
            className="rccs__issuer"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-container-1 MuiGrid-justify-xs-space-between-20"
    style={
      Object {
        "margin": "30px 0",
      }
    }
  >
    <div
      autoFocus={true}
      errors={Object {}}
      fullWidth={true}
      handleChange={[MockFunction]}
      label="Nome impresso no cartão de crédito"
      name="name"
      onFocus={[Function]}
      touched={Object {}}
      type="text"
      value=""
    >
      Field
    </div>
    <div
      errors={Object {}}
      fullWidth={true}
      handleChange={[MockFunction]}
      label="Número do cartão de crédito"
      mask="9999 9999 9999 9999"
      name="number"
      onFocus={[Function]}
      touched={Object {}}
      type="text"
      value=""
    >
      Field
    </div>
    <div
      errors={Object {}}
      handleChange={[MockFunction]}
      label="Data de expiração"
      mask="99/99"
      name="expiry"
      onFocus={[Function]}
      placeholder="MM/AA"
      touched={Object {}}
      type="text"
      value=""
    >
      Field
    </div>
    <div
      errors={Object {}}
      handleChange={[MockFunction]}
      label="Código de segurança"
      mask="9999"
      maskChar=""
      name="cvc"
      onFocus={[Function]}
      placeholder="Ex: 123"
      touched={Object {}}
      type="text"
      value=""
    >
      Field
    </div>
  </div>
  <div
    className="MuiGrid-container-1 MuiGrid-justify-xs-center-18"
  >
    <button
      className="MuiButtonBase-root-124 MuiButton-root-98 MuiButton-contained-109 MuiButton-containedPrimary-110 MuiButton-raised-112 MuiButton-raisedPrimary-113"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="submit"
    >
      <span
        className="MuiButton-label-99"
      >
        Finalizar
      </span>
      <span
        className="MuiTouchRipple-root-127"
      />
    </button>
  </div>
</div>
`;
